qemu/tests/qemu_mem_utilization.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+
+from virttest import utils_test
+from virttest import error_context
+from autotest.client.shared import utils
+from autotest.client.shared import error
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Compare host used memory after guest boot/reboot/shutdown for times:
+    1) boot guest
+    2) after guest up, check qemu memory utilization
+    3) reboot guest for times
+    4) check qemu memory utilization
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def mem_utilization(timeout):
+        cmd = params.get("host_mem")
+        qemu_mem = float(utils.system_output(cmd))
+        end_time = time.time() + float(timeout)
+        while time.time() < end_time:
+            time.sleep(10)
+            cur_qemu_mem = float(utils.system_output(cmd))
+            if qemu_mem == cur_qemu_mem:
+                return qemu_mem
+            qemu_mem = cur_qemu_mem
+        else:
+            raise error.TestFail("Fail to get qemu_mem_utilization")
+
+    timeout = params.get("login_timeout", 120)
+    error_context.context("Check qemu memory utilization before test", logging.info)
+    qemu_mem_utilization_before = mem_utilization(timeout)
+
+    sub_test = params.get("sub_test")
+    vm = env.get_vm(params["main_vm"])
+    error_context.context("Do subtest:%s repeatly" % sub_test, logging.info)
```

Comment:
both 2 variants's sub test are boot, then no difference.
it's better to print the reboot method.